### PR TITLE
feat: add lint for jinja ref

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

### CODE REVIEW

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -641,6 +641,37 @@ def test_jinja2_vars(self):
...
+        with tmp_directory() as recipe_dir:
+            with open(os.path.join(recipe_dir, "recipe.yaml"), "w") as fh:
+                fh.write(
+                    """
+                    package:
+                       name: foo
+                    requirements:
+                      run:
+                        - ${{name}}
+                        - ${{ x.update({4:5}) }}
```
Comment (wolfv) [inline tests/test_lint_recipe.py:660]: Maybe we can lint this as well as this doesn't work with `rattler-build`.

File: tests/test_lint_recipe.py [hunk 2]
```diff
@@ -641,6 +641,37 @@ def test_jinja2_vars(self):
...
+        )
+
+        with tmp_directory() as recipe_dir:
+            with open(os.path.join(recipe_dir, "recipe.yaml"), "w") as fh:
+                fh.write(
+                    """
+                    package:
+                       name: foo
+                    requirements:
+                      run:
```
[the remark this answers is not in the document]
Comment (wolfv) [reply tests/test_lint_recipe.py:658]: @beckermr the test does check that the lines `6, 8, 10, 11, 12` are mentioned. 

Isn't that good enough? I can also add some extra tests.

[the remark this answers is not in the document]
Comment (wolfv) [reply tests/test_lint_recipe.py:658]: It's actually modeled exactly after the test above for the meta.yaml Jinja.

[the remark this answers is not in the document]
Comment (beckermr) [reply tests/test_lint_recipe.py:658]: Ah woops! I missed the specific check. LGTM!

**Pull request merged**